tests/test_dirs.py

```diff
@@ -221,3 +221,39 @@ def test_move(self, reset_db) -> None:
         # new client: force loading from store
         reload_catalog()
         assert pxt.list_tables('dir2') == ['dir2.dir1.sub1.t2']
+
+    def test_create_with_parents(self, reset_db) -> None:
+        all_dirs_3 = ['dir1', 'dir1.dir2', 'dir1.dir2.dir3']
+
+        dir3 = pxt.create_dir('dir1.dir2.dir3', create_parents=True)
+        md = dir3.get_metadata()
+        assert md['path'] == 'dir1.dir2.dir3'
+        assert md['name'] == 'dir3'
+        listing = pxt.list_dirs(recursive=True)
+        assert listing == all_dirs_3
+
+        pxt.drop_dir('dir1.dir2.dir3')
+        pxt.drop_dir('dir1.dir2')
+
+        dir3 = pxt.create_dir('dir1.dir2.dir3', create_parents=True)
+        md = dir3.get_metadata()
+        assert md['path'] == 'dir1.dir2.dir3'
+        assert md['name'] == 'dir3'
+        listing = pxt.list_dirs(recursive=True)
+        assert listing == all_dirs_3
+
+        pxt.drop_dir('dir1.dir2.dir3')
```

Comment:
How is this different than the last scenario?

Reply:
changed it only 2 cases there create entire path,
create partial path